subjects/road_intersection/README.md

```diff
@@ -7,71 +7,106 @@ Its up to you to decide which library and file system you want do use to create
+           r0 ↓          |
+_______________          |          _____________
+               |         | ←  ↑  → |
+               |    ↓    |r2 r1 r0 |
+               |    ↓    |    ↑    |
+               |    ↓    |    ↑    |
+               |    ↓    |    ↑    |
+               |    ↓    |    ↑    |
+               |    ↓    |    ↑    |
 ```
 
-There are 4 incoming lanes. Each lane has a set of consecutive unique ids `(ri, ri, ri)` where `(i = 0, 1, ⋯, 11)`, numbered clockwise starting from the top-most. Each set of ids
-indicates the outgoing direction of a vehicle. Vehicles driving in a given lane will have a given **route** (r0 | r1 .... r11) that indicates the lane destination of that vehicles.
+For clarification reasons we will assume that a lane can have three different routes (consider you are in the vehicle position):
+
+- `r0`, turning right
+- `r1`, straight ahead
+- `r2`, turning left
+
+2. Traffic lights
 
-For this raid you must follow these assumptions:
+Traffic lights are signalling devices positioned at road intersections that follows an universal color code,
+normally its green, red and amber, but for this project you will just use the colors **red** and **green**.
 
-- Vehicles are to be **autonomous**, so Vehicles driving on a lane with a **given route** must follow the direction of that route, its not possible for the
-  driver to change lane.
+You will then have to create some kind of representation for the traffic lights and distribute them for each lane in the intersection.
 
-- Other traffic units such as pedestrians and emergency vehicles are not considered.
+You are free to decide what algorithm you want to implement for the traffic light system, but keep in mind that traffic congestion should not be to high.
 
-- Vehicles can have different velocities, it is up to you to decide the velocity for each vehicle but it must be at least two different
-  velocities (0 does not count as a velocity for the vehicle).
+3. Vehicles
 
-- Vehicles should not collide in a traffic lane. It should be kept a safety distance from other vehicles
+```
+  ______
+ /|_||_\`.__
+=`-(_)--(_)-'
+```
 
-- Vehicles should be generated randomly, both route and lane should be also random.
+Vehicles must obey this rules:
 
-#### **Stop conditions**
+- Autonomous, vehicles driving on a lane with a **given route** must 
+follow the direction of that route, its not possible for the driver to change lanes or route.
 
-As you know vehicles must not collide, but for **debugging and testing** reasons you must show some kind of feedback (logs, terminal display or canvas display its up to you to decide)
-if vehicles collide, so that the users can conclude that the vehicles collided.
-Each lane will have a traffic light system (either imaginary of real), the objective is that each vehicle stops if there is any type of collision with other vehicles. Otherwise the vehicles can proceed in their lane.
+- Each vehicle must have a fixed velocity.
```

Comment:
Maybe we can add a distinction for the car so we can see the route that they are suppose to follow. Maybe, a different color for each route. Like red cars turn left. Or something like that. So the reviewer can check that the car is indeed following the correct route.

Reply:
about the colors i agree. we can let them choose the colors they want, but it was to be different colors for each route